### PR TITLE
4.next: middleware DIC

### PR DESCRIPTION
**Pull request by LordSimal**
Thanks to @ADmad for providing the patch.

### CODE REVIEW

**Comment by ADmad** [comment]
Tests required :slightly_smiling_face:

File: tests/TestCase/Http/MiddlewareQueueTest.php [hunk 1]
```diff
@@ -390,4 +391,13 @@ public function testAddingDeprecatedDoublePassMiddleware(): void
...
         });
     }
+
+    public function testDIContainer(): void
+    {
+        $container = new Container();
+        $container->add(SampleMiddleware::class, new SampleMiddleware());
+        $queue = new MiddlewareQueue([], $container);
+        $queue->add(SampleMiddleware::class);
+        $this->assertInstanceOf(SampleMiddleware::class, $queue->current());
```
Comment (ADmad) [inline tests/TestCase/Http/MiddlewareQueueTest.php]: You need to assert it's the same instance set to the container. Instanceof check will pass even if the instance was created within the middleware queue.

Comment (LordSimal) [reply tests/TestCase/Http/MiddlewareQueueTest.php]: Right, didn't think of that 👍🏻

File: src/Http/MiddlewareQueue.php [hunk 2]
```diff
@@ -70,14 +78,22 @@ public function __construct(array $middleware = [])
...
-            if ($className === null) {
-                throw new RuntimeException(sprintf(
-                    'Middleware "%s" was not found.',
-                    $middleware
-                ));
+            if ($this->container && $this->container->has($middleware)) {
+                $middleware = $this->container->get($middleware);
+            }
+
+            if (is_string($middleware)) {
```
Comment (markstory) [inline src/Http/MiddlewareQueue.php]:
```suggestion
            } else {
```

Seems like this condition is redundant if we use `else`.

**Pull request merged**